server/db_accessor.cpp

```diff
@@ -4,59 +4,125 @@
 #include <openssl/sha.h>
 
 using namespace nibaserver;
+using namespace ozo::literals;
+namespace sev = boost::log::trivial;
 
-std::unordered_map<std::string, db_accessor::user> db_accessor::db_;
 std::unordered_map<std::string, nibashared::character> db_accessor::char_tbl_;
 
 constexpr std::size_t HASH_SIZE = 32;
+static auto conn_info = ozo::make_connection_info("host=127.0.0.1 port=5432 dbname=niba");
+nibaserver::logger db_accessor::logger_ = logger();
+
+bool nibaserver::db_accessor::login(boost::asio::io_context &ioc, boost::asio::yield_context &yield,
+    const std::string &id, const std::string &password) {
+    ozo::error_code ec{};
+    ozo::rows_of<ozo::pg::name, std::vector<char>, std::vector<char>, bool> user_credential;
+    auto conn = ozo::request(ozo::make_connector(conn_info, ioc),
```

Comment:
I am thinking of passing in (or create one on server start up) a connection_pool or a  connection provider that the db_accessor has direct access to.
https://github.com/yandex/ozo/blob/master/examples/connection_pool.cpp
So the processor and session don't really need to know anything about `io_context`. And we only pass around the `yield_context`

Reply:
thinking about it, we could have 1 connector (from a pool) per db_accessor, and 1 db_accessor per session, and give the processor a reference to the db_accessor instead

Reply:
Done in the latest commit. Create the connection pool in main. Initialize the per-session db_accessor. Pass the connector into  db_accessor.